src/dom/react-nodegui/src/components/Svg/RNSvg.ts

```diff
@@ -0,0 +1,218 @@
+   */
+  nodeParent?: Component;
+
+  /**
+   * Sets the stylesheet
+   */
+  styleSheet?: string;
+
+  /**
+   * Sets the file path
+   */
+  src?: string | Buffer;
+}
+
+const setSvgProps = (
+  widget: RNSvg,
+  newProps: SvgProps,
+  oldProps: SvgProps
+) => {
+  const setter: SvgProps = {
+    set acceptDrops(on: boolean) {
+      widget.setAcceptDrops(on);
+    },
+    set cursor(cursor: CursorShape|QCursor) {
+      widget.setCursor(cursor);
+    },
+    set enabled(enabled: boolean) {
+      widget.setEnabled(enabled);
+    },
+    set fixedSize({width, height}: {width: number, height: number}) {
+      widget.setFixedSize(width, height);
+    },
+    set flexNodeSizeControlled(isSizeControlled: boolean) {
+      widget.setFlexNodeSizeControlled(isSizeControlled);
+    },
+    set font(font: QFont) {
+      widget.setFont(font);
+    },
+    set geometry({x, y, w, h}: {x: number, y: number, w: number, h: number}) {
+      widget.setGeometry(x, y, w, h);
+    },
+    set graphicsEffect(effect: QGraphicsEffect<any>) {
+      widget.setGraphicsEffect(effect);
+    },
+    set inlineStyle(style: string) {
+      widget.setInlineStyle(style);
+    },
+    set layout(parentLayout: NodeLayout<QWidgetSignals>) {
+      widget.setLayout(parentLayout);
+    },
+    set maximumSize({maxw, maxh}: {maxw: number, maxh: number}) {
+      widget.setMaximumSize(maxw, maxh);
+    },
+    set minimumSize({minw, minh}: {minw: number, minh: number}) {
+      widget.setMinimumSize(minw, minh);
+    },
+    set mouseTracking(isMouseTracked: boolean) {
+      widget.setMouseTracking(isMouseTracked);
+    },
+    set nodeParent(parent: Component) {
+      widget.setNodeParent(parent);
+    },
+    set styleSheet(styleSheet: string) {
+      widget.setStyleSheet(styleSheet);
+    },
+    set src(file: string | Buffer) {
+      widget.load(file);
+    },
+    set id(id: string) {
+      widget.setObjectName(id);
+    },
+    set on(
+      listenerMap: Partial<WidgetEventListeners | QWidgetSignals>
+    ) {
+      const listenerMapLatest: any = Object.assign({}, listenerMap);
+      const oldListenerMap = Object.assign({}, oldProps.on);
+      Object.entries(oldListenerMap).forEach(([eventType, oldEvtListener]) => {
+        const newEvtListener = listenerMapLatest[eventType];
+        if (oldEvtListener !== newEvtListener) {
+          widget.removeEventListener(eventType as any, oldEvtListener);
+        } else {
+          delete listenerMapLatest[eventType];
+        }
+      });
+
+      Object.entries(listenerMapLatest).forEach(
+        ([eventType, newEvtListener]) => {
+          widget.addEventListener(eventType as any, newEvtListener);
+        }
+      );
+    },
+    set visible(shouldShow: boolean) {
+      shouldShow ? widget.show() : widget.hide();
+    },
+  };
+  Object.assign(setter, newProps);
+};
+
+/**
+ * @ignore
+ */
+export class RNSvg extends QSvgWidget implements RNComponent {
+  static tagName = "svg";
```

Comment:
I'm unsure how the Svelte compiler will handle the tag name `svg` – I'm worried that it might special-case it. Have you tried running the code and checking what the Intellisense looks like when you try to write attributes on an `<svg>` element? If they appear to clash with those of HTML elements, we may have to rename this to `<qtsvg>` or something to avoid it.